apps/dashboard/src/pages/Invoices.tsx

```diff
@@ -1,52 +1,25 @@
 import List from "@/components/Invoices/List";
-import {
-  Button,
-  ButtonSize,
-  Loading,
-  useFormatMessage,
-  usePaginationQuery,
-  atlas,
-} from "@litespace/luna";
-import { isEmpty } from "lodash";
-import React, { useCallback } from "react";
+import { useFindInvoices } from "@litespace/headless/invoices";
+import { Loading, useFormatMessage } from "@litespace/luna";
+import React from "react";
 
 const Invoices: React.FC = () => {
   const intl = useFormatMessage();
-  const findInvoices = useCallback(
-    async ({ pageParam }: { pageParam: number }) => {
-      return await atlas.invoice.find({ page: pageParam });
-    },
-    []
-  );
-
-  const { list, query, more } = usePaginationQuery(findInvoices, [
-    "find-invoices",
-  ]);
-
-  const onUpdate = useCallback(() => {
-    query.refetch();
-  }, [query]);
+  const { list, query } = useFindInvoices({ pageParam: 1 });
+  console.log(query);
 
   return (
-    <div className="w-full max-w-screen-2xl mx-auto p-8">
+    <div className="w-full p-8 mx-auto max-w-screen-2xl">
       <div className="mb-4">
         <h1 className="text-3xl">{intl("invoices.title")}</h1>
       </div>
-
-      <Loading className="h-screen" show={query.isLoading} />
-      {list && !isEmpty(list) ? (
-        <div className="w-full">
-          <List invoices={list} onUpdate={onUpdate} />
-          <Button
-            size={ButtonSize.Small}
-            onClick={more}
-            loading={query.isLoading || query.isFetching}
-            disabled={query.isLoading || query.isFetching || !query.hasNextPage}
-          >
-            {intl("global.labels.more")}
-          </Button>
-        </div>
-      ) : null}
+      <div className="w-full">
+        {query.isLoading ? (
+          <Loading className="h-screen" show={query.isLoading} />
+        ) : (
+          <List invoicesList={list} query={query} />
+        )}
```

Comment:
Always remember to handle the error state as well. Please take a look at the users page to see how we are handling the errors.

Reply:
https://github.com/litespace-org/litespace/blob/0795a725442639fd1ef03bb212a4304766f8577b/apps/dashboard/src/components/Plans/List.tsx#L155-L162